### PR TITLE
Feat/admin module core

### LINKED ISSUE #1507: Admin Panel: Create routes & navigation
## Description
Part of the larger effort to create an Admin Panel to allow admin users to manage content from within the platform itself. Overarching issue to come.

Create the protected routes within the `/admin` section of the site, a secondary navigation bar, and a primary menu item to reach the `/admin` page.

Acceptance Criteria Checklist:
- [x] Users with the `admin` user role can access a new Admin menu item within the user menu.
- [x] A new `/admin` directory that is protected to only allow users with the `admin` role to access it. Other users can be redirected to the home page.
- [x] Pages for each of the main content types to be managed within the Admin Panel, in addition to an Approvals page. Note that for now these pages can just have an H1 with their page title as a placeholder:
  - [x] Approvals: `/admin`
  - [x] Browse Users: `/admin/users`
  - [ ] Edit User: `/admin/users/[id]`
  - [x] Browse Map Pins: `/admin/map-pins`
  - [x] Browse Tags: `/admin/tags`
  - [x] Browse How To's: `/admin/how-tos`
  - [ ] Edit How To's: `/admin/how-tos/[id]`
  - [x] Browse Research: `/admin/research`
  - [ ] Edit Research: `/admin/research/[id]`
- [x] There is a secondary nav that appears on any of the pages above.
- [x] The current subdirectory/page is highlighted within the secondary nav

 ## Related Pages
Related to all other pages within the larger Admin Panel effort (overarching issue to come).

## Mockups
[Visit the Figma space](https://www.figma.com/file/nPDwFo0Ee4wcHvtnEGtAu9/?node-id=10%3A74) where the mockups are hosted to fully inspect them. Comments and questions about them can be added by pressing C and clicking where you want to pin a discussion, tag @franknoirot or @davehakkens within.

Here are screenshots of the 2 relevant desktop designs as of 28/02/22:

1. A view of the primary nav expanded, showing where the Admin menu link should be included if a user has the Admin role.
Pending Approvals - Top Nav Menu Expanded

2. The browse view of all Users, which shows the secondary navigation with the Users link highlighted as the current page is within that directory.
Users Browse

### PR DESCRIPTION
**Pull request by chrismclarke**
PR Checklist

- [x] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)
- [x] - Latest `master` branch merged

PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Developer experience (improves developer workflows for contributing to the project)

## Description

- Scaffold out admin module
- Add base routes (with auth restrictions) and placeholder components
- Minor docs update for adding new modules
- Minor code tidying to allow auth-only pages in settings dropdown and auto-redirect for pages where user cannot access

## Git Issues

closes #1507

## Screenshots/Videos
_Update 2022-03-03 - menu moved to profile dropdown and redirect added for non-authenticated users_

https://user-images.githubusercontent.com/10515065/156623295-615e22cf-eabe-44b1-96fb-b9f690c6e161.mp4

_Original_

https://user-images.githubusercontent.com/10515065/156493404-4ec48cb5-cc54-448a-b994-35e7b88ba1b2.mp4

---

## What happens next?

Thanks for the contribution! We try to make sure all PRs are reviewed ahead of a monthly dev call (first Monday of the month, open to all!).

If the PR is working as intended it'll be merged and included in the next platform release, if not changes will be requested and re-reviewed once updated.

If you need more immediate feedback you can try reaching out on slack in the `platform-dev` channel.

### CODE REVIEW

File: src/modules/index.test.ts [hunk 1]
```diff
@@ -1,61 +1,55 @@
...
+    process.env.REACT_APP_SUPPORTED_MODULES = ''
+    expect(getSupportedModules()).toStrictEqual([
+      MODULE.CORE,
+      MODULE.HOWTO,
+      MODULE.MAP,
+      MODULE.EVENTS,
+      MODULE.RESEARCH,
+      MODULE.ACADEMY,
+      MODULE.USER,
+      MODULE.ADMIN,
```
Comment (chrismclarke) [inline src/modules/index.test.ts:18]: Only actual change is to add admin module to list of supported - the rest is just prettier auto-formatting

File: src/modules/index.ts [hunk 2]
```diff
@@ -1,23 +1,29 @@
...
-    ACADEMY = 'academy',
-    USER = 'user',
+  CORE = 'core', // This is enabled on all installations
+  HOWTO = 'howto',
+  MAP = 'map',
+  EVENTS = 'events',
+  RESEARCH = 'research',
+  ACADEMY = 'academy',
+  USER = 'user',
+  ADMIN = 'admin_v2',
```
Comment (chrismclarke) [inline src/modules/index.ts:11]: for now we are calling the new module `admin_v2` to allow the legacy admin to still be used until ready to switch over

File: src/components/ProfileModal/ProfileModal.tsx [hunk 3]
```diff
@@ -87,17 +88,15 @@ export class ProfileModal extends React.Component<IProps> {
...
-                key={page.path}
-                to={page.path}
-                data-cy={`menu-${page.title}`}
-              >
-                <Flex>{page.title}</Flex>
-              </ModalLink>
-            ))}
-          </Flex>
+          {COMMUNITY_PAGES_PROFILE.map(page => (
+            <AuthWrapper roleRequired={page.requiredRole} key={page.path}>
```
Comment (chrismclarke) [inline src/components/ProfileModal/ProfileModal.tsx:92]: Additional wrapper added to allow additional links in profile dropdown for auth-required pages

**Review by davehakkens — APPROVED** [review]

**Pull request merged**